### PR TITLE
ARROW-14754: [Doc] Steps in making your first PR - building R package

### PR DESCRIPTION
**Pull request by thisisnic**

### CODE REVIEW

**Comment by thisisnic** [comment]
@AlenkaF Please can you review this and let me know if this is the kind of thing you'd like in this section or if you'd prefer I add more detail?

**Comment by AlenkaF** [comment]
@thisisnic I think it is great! Good intro to the main part of the documentation.

I did get an idea when looking at this PR to move notes about [recompiling cython, C++ ](https://github.com/apache/arrow/blob/master/docs/source/developers/guide/step_by_step/testing.rst) from testing section to this part of the guide. I think it fits better. And then a sentence or two could be added for the R package, that it needs no recompiling (if I understand correctly).

What do you think? I could try to push a commit with the suggestion here?

**Comment by thisisnic** [comment]
> @thisisnic I think it is great! Good intro to the main part of the documentation.
> 
> I did get an idea when looking at this PR to move notes about [recompiling cython, C++ ](https://github.com/apache/arrow/blob/master/docs/source/developers/guide/step_by_step/testing.rst) from testing section to this part of the guide. I think it fits better. And then a sentence or two could be added for the R package, that it needs no recompiling (if I understand correctly).
> 
> What do you think? I could try to push a commit with the suggestion here?

I think it makes more sense here, though you could always mention it in both places if it fits in both? Maybe here could be the detailed version, and in the testing section you just briefly remind the reader they should recompile?

**Comment by AlenkaF** [comment]
Makes sense, thank you! Will make it as a separate PR for the rebuilding topic.
As for this PR, I have no comments 👍

File: docs/source/developers/guide/step_by_step/building.rst [hunk 1]
```diff
@@ -150,5 +150,13 @@ interested to find more information.
...
          - :ref:`build_pyarrow_win`
 
-   .. tab:: Building the R package
 
+   .. tab:: Building the R package
+      
+	  When working on code in the R package, you only need to build the Arrow C++ 
+	  library (often referred to in the R documentation as 'libarrow') from source 
+	  if you wish to make changes to the code there; otherwise you can install a 
+	  binary version of the C++ libary and then need to link it up with the R package.
```
Comment (dragosmg) [inline docs/source/developers/guide/step_by_step/building.rst:159]: I think this still doesn't work - installing `libarrow` from homebrew (on Mac) and having only the R package as `source`.

Comment (thisisnic) [reply docs/source/developers/guide/step_by_step/building.rst:159]: Great point, will update

**Pull request closed without merging**